### DISCUSSION TITLE
Optional vs Required Rules

### DISCUSSION THREAD

**Discussion by aka-tpayne** [rectorphp/rector#6757] [General]
I didn't see anything already in the codebase, but I think optional vs required rules would be a great addition.

As great as some of the rules are, some of them just clutter up the changes, like TernaryToElvisRector.

Obviously, I can go and read through each and every rule and decide for myself, but it would be nice to have different sets for actually required rules that must be run to say upgrade to a new PHP version.

Thoughts?

**Comment by TomasVotruba** [comment]
Hi, thanks the idea. While this might be helpful for this one specific set, it will bring DX hit.
There is already enough small sets and more fractioning would make their usage very exhausting:

```bash
SetList::PHP_80
SetList::PHP_80_REQUIRED
SetList::PHP_80_DEPRECATED
SetList::PHP_80_WARNINGS
...
```

Most people want to use Rector to upgrade their code base, so they can work on their own domain. The more work and responsibility Rector resolves, the better. 

Yet, if you need such rulesets and you believe they have value in the community, you can create a custom repository and share it here :+1:
